lib/core/versions/latest/util/JsObject.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Class containing JavaScript object operations.
+ */
+export default class JsObject {
+  /**
+   * Deep copies the given input.
+   */
+  public static deepCopyObject (input: any): any {
```

Comment:
Maybe it's good to have specific tests for this. Something simple like

```
const a = {number: 1, object: {x: 'x'}, array: [1,2,3]};
const b = deepCopyObject(a);

a.number = 2;
a.object.x = 'y';
a.array[0] = 4;

expect(a.number).not.toEqual(b.number);
expect(a.object).not.toEqual(b.object);
expect(a.array).not.toEqual(b.array);
```

But this is indirectly tested so I'm ok if you don't want to add it.

Reply:
Sure, I'll add it.